### PR TITLE
📚Switch to using DocumenterInterLinks for ManifoldsBase.jl 

### PR DESCRIPTION
**Pull request by kellertuer**
This is still a bit WIP, since we have to 

* [x] check all occurrences of `.io/ManifoldsBase.jl/` in the docs and check whether to replace that with an interlink (probably in all cases besides the Readme.md
* [x] figure out how to handle that in the .qmd files

But in general it seems to work really nice and smoothly, following the ideas mentioned at http://juliadocs.org/DocumenterInterLinks.jl/stable/syntax/

### CODE REVIEW

**Comment by kellertuer** [comment]
This should be finished so far. I think since this is just a technical update, it should maybe not get its own release but be part of the next one.

File: Changelog.md [hunk 1]
```diff
@@ -5,6 +5,12 @@ All notable Changes to the Julia package `Manopt.jl` will be documented in this
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [unreleased] March 15, 2024
+
+### Changed
+
+* `ManifoldsBase.jl` now is running on Documenter 1.3, `Manopt.jl` documentation now uses [DocumenterInterLinks]() to refer to sections and functions from `ManifoldsBasel.jl`
```
Comment (mateuszbaran) [inline Changelog.md:12]:
```suggestion
* `ManifoldsBase.jl` now is running on Documenter 1.3, `Manopt.jl` documentation now uses [DocumenterInterLinks](https://github.com/JuliaDocs/DocumenterInterLinks.jl) to refer to sections and functions from `ManifoldsBase.jl`
```

**Comment by mateuszbaran** [comment]
LGTM.

Do you prefer do merge this one first or the sectional curvature one?

**Comment by kellertuer** [comment]
Yes, I think merging this one first has two advantages

* you can use the same new scheme directly to link to the sectional curvature as well 
* merging this into your PR hopefully also fixes the documentation run there.

**Review by mateuszbaran — APPROVED** [review]

**Pull request merged**